src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -143,83 +86,15 @@ public FileUploadSubmission handleFileUploadSubmission(FileUploadSubmission file
-     * @return the newly created result
-     */
-    public Result setNewResult(FileUploadSubmission submission) {
-        Result result = new Result();
-        result.setSubmission(submission);
-        submission.setResult(result);
-        if (submission.getParticipation() != null) {
-            submission.getParticipation().addResult(result);
-        }
-        resultRepository.save(result);
-        fileUploadSubmissionRepository.save(submission);
-        return result;
-    }
-
-    /**
-     * Saves the given submission. Is used for creating and updating file upload submissions. Rolls back if inserting fails - occurs for concurrent createFileUploadSubmission() calls.
-     *
-     * @param fileUploadSubmission the submission that should be saved
-     * @param file                 the file that will be saved on the server
-     * @param participation        the participation the submission belongs to
-     * @param exercise             the exercise the submission belongs to
-     * @return the fileUploadSubmission entity that was saved to the database
-     * @throws IOException if file can't be saved
-     */
-    @Transactional(rollbackFor = Exception.class)
-    public FileUploadSubmission save(FileUploadSubmission fileUploadSubmission, MultipartFile file, StudentParticipation participation, FileUploadExercise exercise)
-            throws IOException {
-        final var localPath = saveFileForSubmission(file, fileUploadSubmission, exercise);
-
-        // update submission properties
-        fileUploadSubmission.setSubmissionDate(ZonedDateTime.now());
-        fileUploadSubmission.setType(SubmissionType.MANUAL);
-        fileUploadSubmission.setParticipation(participation);
-        fileUploadSubmission = fileUploadSubmissionRepository.save(fileUploadSubmission);
-        fileUploadSubmission.setFilePath(fileService.publicPathForActualPath(localPath, fileUploadSubmission.getId()));
-        fileUploadSubmissionRepository.save(fileUploadSubmission);
-
-        participation.addSubmissions(fileUploadSubmission);
-
-        if (fileUploadSubmission.isSubmitted()) {
-            participation.setInitializationState(InitializationState.FINISHED);
-
-            messagingTemplate.convertAndSendToUser(participation.getStudent().getLogin(), "/topic/exercise/" + participation.getExercise().getId() + "/participation",
-                    participation);
-        }
-        StudentParticipation savedParticipation = studentParticipationRepository.save(participation);
-        if (fileUploadSubmission.getId() == null) {
-            Optional<FileUploadSubmission> optionalFileUploadSubmission = savedParticipation.findLatestFileUploadSubmission();
-            if (optionalFileUploadSubmission.isPresent()) {
-                fileUploadSubmission = optionalFileUploadSubmission.get();
-            }
-        }
-
-        return fileUploadSubmission;
+        return getRandomUnassessedSubmission(fileUploadExercise, FileUploadSubmission.class);
```

Comment:
This method can also be moved to a superclass.

Reply:
You are right. Moved it